### PR TITLE
Fighters: code cleanup

### PR DESCRIPTION
**Pull request by Cjkjvfnby**

### CODE REVIEW

File: default/AI/CombatRatingsAI.py [hunk 1]
```diff
@@ -45,26 +44,20 @@ class ShipCombatStats(object):
...
-            if stat_tuple and isinstance(stat_tuple, tuple):
-                attack_tuple, self.structure, self.shields = stat_tuple
-                self.attacks = tuple_to_dict(attack_tuple)
-            else:
-                self.structure = structure
-                self.shields = shields
-                self.attacks = attacks or {}
+            self.structure = 1 if structure is None else structure
+            self.shields = 0 if shields is None else shields
+            self.attacks = {} if attacks is None else {}
```
Comment (Morlic-fo) [inline default/AI/CombatRatingsAI.py]: Should probably be `else attacks`

File: default/AI/CombatRatingsAI.py [hunk 2]
```diff
@@ -275,26 +271,28 @@ def get_fleet_rating(fleet_id, enemy_stats=None):
...
-    """Get species modifier.
+@cache_by_session
+def _get_species_grades(species_name, grade_type):
+    spec_tags = []
+    if species_name:
+        species = fo.getSpecies(species_name)
+        if species:
+            spec_tags = species.tags
+        else:
+            print "Error: get_species_grades couldn't retrieve species '%s'" % species_name
```
Comment (Morlic-fo) [inline default/AI/CombatRatingsAI.py]: probably should print to stderr instead?

File: default/AI/CombatRatingsAI.py [hunk 3]
```diff
@@ -275,26 +271,28 @@ def get_fleet_rating(fleet_id, enemy_stats=None):
...
-            if species:
-                spec_tags = species.tags
-            else:
-                print "Error: get_piloting_grades couldn't retrieve species '%s'" % species_name
-        piloting_grades[species_name] = (get_ai_tag_grade(spec_tags, 'WEAPONS'),
-                                         get_ai_tag_grade(spec_tags, 'SHIELDS'),
-                                         get_ai_tag_grade(spec_tags, 'ATTACKTROOPS'),
-                                         )
-    return piloting_grades[species_name]
+def get_pilot_weapons_grade(species_name):
```
Comment (Morlic-fo) [inline default/AI/CombatRatingsAI.py:288]: docstring return type please.

**Comment by Vezzra** [comment]
As the Fighters branch has been rebased, this one will need to be rebased too I guess...

**Comment by Cjkjvfnby** [comment]
I rebased my branch and add fixes to corresponding commits.

**Comment by geoffthemedio** [comment]
any further concerns?

**Comment by Cjkjvfnby** [comment]
This branch missed couple of rebases of its parent, I will rebase it after release.

**Comment by Vezzra** [comment]
> This branch missed couple of rebases of its parent, I will rebase it after release.

So I guess whoever feels dedicated to that task can resume this now?

**Comment by adrianbroher** [comment]
I don't get quite the purpose of this PR. Are those fixes that should be applied onto `freeorion/Fighters`?

**Comment by adrianbroher** [comment]
Applied relevant changes into Fighters branch.

**Pull request closed without merging**